src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,289 @@
+        { value: "ordered-puzzle", label: "Ordered Puzzle" },
+    ];
+
+    useEffect(() => {
+        const getPuzzleBody = async () => {
+            const data = await getPuzzle(sceneId, objectId, handleError);
+            setAnimationsJson(data.animations_json);
+            if (Object.keys(data.animations_json).length === 0) {
+                setPuzzleBody({ jsonData: {} });
+                setIsInteractable(false);
+            } else {
+                setPuzzleBody(data.animations_json.blackboardData);
+                setPuzzleType(
+                    data.animations_json.blackboardData.componentType
+                );
+                setIsInteractable(data.is_interactable);
+            }
+        };
+        if (Object.keys(puzzleBody).length === 0) {
+            getPuzzleBody();
+        }
+    }, [sceneId, objectId, puzzleBody, handleError]);
+
+    const selectPuzzleType = (obj) => {
+        if (obj) {
+            const data = puzzleBody;
+            data.componentType = obj.value;
+            if (obj.value === "text-pane") {
+                if (!data.jsonData.data) {
+                    data.jsonData.data = [];
+                }
+            }
+            setPuzzleBody(data);
+            setPuzzleType(obj.value);
+        }
+    };
+
+    const saveTexts = (texts) => {
+        const animCopy = animationsJson;
+        const puzzleBodyCopy = puzzleBody;
+        puzzleBodyCopy.jsonData.data = texts;
+        animCopy.blackboardData = puzzleBodyCopy;
+        setAnimationsJson(animCopy);
+        handleSave();
+    };
+
+    const handleSave = () => {
+        const savePuzzle = async () => {
+            await editPuzzle(
+                { sceneId, objectId, isInteractable, animationsJson },
+                handleError
+            );
+        };
+        savePuzzle();
+    };
+
+    const toggleButton = () => {
+        setIsInteractable(!isInteractable);
+    };
+
+    return (
+        <div className={classes.container}>
+            <div>
+                <label htmlFor="subscribeNews">Interactable?</label>
+                <input
+                    type="checkbox"
+                    value={isInteractable}
+                    checked={isInteractable}
+                    onChange={toggleButton}
+                ></input>
+            </div>
+            {isInteractable ? (
+                <Select
+                    value={puzzleTypeList.filter(
+                        (option) => option.value === puzzleType
+                    )}
+                    options={puzzleTypeList}
+                    placeholder="Select puzzle type..."
+                    noResultsText="No puzzle types found"
+                    searchable={true}
+                    onChange={selectPuzzleType}
+                />
+            ) : null}
+            {isInteractable && puzzleType === "text-pane" ? (
+                <TextPaneView
+                    saveTexts={saveTexts}
```

Comment:
To answer your question Amolik, this here will work even if TextPaneView is defined in a different file. You are passing the function, they don't need to be defined in same file.

Let's move TextPaneView to a new file.